packages/nodes-base/nodes/Merge/v2/MergeV2.node.ts

```diff
@@ -605,8 +604,36 @@ export class MergeV2 implements INodeType {
 					returnData.push.apply(returnData, this.getInputData(1));
 				}
 				if (output === 'empty') {
-					const itemData = generatePairedItemData(this.getInputData(0).length);
-					returnData.push({ json: {}, pairedItem: itemData });
+					// TODO: was necessary to satisfies types.
+					// I'm wondering if there is already a function that does exactly that? That turns
+					// Array<IPairedItemData | IPairedItemData[] | number | undefined> -> IPairedItemData[]
+					// or
+					// IPairedItemData | IPairedItemData[] | number | undefined -> IPairedItemData | undefined
+					function numberToPairedItem(
+						possiblyPairedItems: Array<INodeExecutionData['pairedItem']>,
+					): IPairedItemData[] {
+						return possiblyPairedItems.reduce((pairedItems: IPairedItemData[], pairedItem) => {
+							if (typeof pairedItem === 'number') {
+								pairedItems.push({ item: pairedItem });
+							} else if (Array.isArray(pairedItem)) {
+								pairedItems.push(...pairedItem);
+							} else if (pairedItem) {
+								pairedItems.push(pairedItem);
+							}
+
+							return pairedItems;
+						}, [] as IPairedItemData[]);
+					}
```

Comment:
@despairblue I guess if this is something that may come up again it would be handy to have a utility function for it so we can reuse it but I am not sure

Reply:
@Joffcom There was a function already that does this, just had to combine it with a flatMap :)